library/std/src/sys/backtrace.rs

```diff
@@ -68,61 +68,67 @@ unsafe fn _print_fmt(fmt: &mut fmt::Formatter<'_>, print_fmt: PrintFmt) -> fmt::
                 return false;
             }
 
-            let mut hit = false;
-            backtrace_rs::resolve_frame_unsynchronized(frame, |symbol| {
-                hit = true;
-
-                // `__rust_end_short_backtrace` means we are done hiding symbols
-                // for now. Print until we see `__rust_begin_short_backtrace`.
-                if print_fmt == PrintFmt::Short {
-                    if let Some(sym) = symbol.name().and_then(|s| s.as_str()) {
-                        if sym.contains("__rust_end_short_backtrace") {
-                            print = true;
-                            return;
-                        }
-                        if print && sym.contains("__rust_begin_short_backtrace") {
-                            print = false;
-                            return;
-                        }
-                        if !print {
-                            omitted_count += 1;
+            if cfg!(feature = "backtrace-trace-only") {
+                const HEX_WIDTH: usize = 2 + 2 * core::mem::size_of::<usize>();
+                let frame_ip = frame.ip();
+                res = writeln!(bt_fmt.formatter(), "{idx:4}: {frame_ip:HEX_WIDTH$?}");
```

Comment:
To be able to actually symbolize the backtraces this would need to print both the path of the DSO (executable or dylib) and the offset within this DSO. It could be a dylib loaded at runtime and thanks to ASLR the exact location of DSOs can't be determined anymore after the fact unless we print it here.

Reply:
This could take inspiration on what backtrace-rs does on Fuchsia which unconditionally does backtrace symbolication out of process.

Reply:
Yeah, I don't think the backtraces are all that usable without symbols when ASLR is disabled. I guess it still shows symbol names for system libraries.

Reply:
Ok, I'm going to punt on doing the hard thing again 🙂. This is (arguably) a pre-existing issue and at least somewhat platform specific so would be much better fixed in backtrace-rs than std. This feature can then use whatever backtrace-rs has once that's written.